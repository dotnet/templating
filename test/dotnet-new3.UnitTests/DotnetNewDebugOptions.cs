// Licensed to the .NET Foundation under one or more agreements.
// The .NET Foundation licenses this file to you under the MIT license.

#nullable enable

using FluentAssertions;
using Microsoft.NET.TestFramework.Assertions;
using Microsoft.TemplateEngine.TestHelper;
using Xunit;
using Xunit.Abstractions;

namespace Dotnet_new3.IntegrationTests
{
    public class DotnetNewDebugOptions
    {
        private readonly ITestOutputHelper _log;

        public DotnetNewDebugOptions(ITestOutputHelper log)
        {
            _log = log;
        }

        [Fact]
        public void CanShowBasicInfoWithDebugReinit()
        {
            string home = TestUtils.CreateTemporaryFolder("Home");
            string cacheFilePath = Path.Combine(home, "dotnetcli-preview", "v2.0.0", "templatecache.json");

            var commandResult = new DotnetNewCommand(_log)
                .WithCustomHive(home)
                .Execute();

            commandResult.Should().ExitWith(0).And.NotHaveStdErr();
            Assert.True(File.Exists(cacheFilePath));
            DateTime lastUpdateDate = File.GetLastWriteTimeUtc(cacheFilePath);

            var reinitCommandResult = new DotnetNewCommand(_log, "--debug:reinit")
               .WithCustomHive(home)
               .Execute();

            reinitCommandResult.Should().ExitWith(0).And.NotHaveStdErr();
            Assert.Equal(commandResult.StdOut, reinitCommandResult.StdOut);
            Assert.True(File.Exists(cacheFilePath));
            Assert.True(lastUpdateDate < File.GetLastWriteTimeUtc(cacheFilePath));
        }

        [Fact]
        public void CanShowBasicInfoWithDebugRebuildCache()
        {
            string home = TestUtils.CreateTemporaryFolder("Home");
            string cacheFilePath = Path.Combine(home, "dotnetcli-preview", "v2.0.0", "templatecache.json");

            var commandResult = new DotnetNewCommand(_log)
                .WithCustomHive(home)
                .Execute();

            commandResult.Should().ExitWith(0).And.NotHaveStdErr();
            Assert.True(File.Exists(cacheFilePath));
            DateTime lastUpdateDate = File.GetLastWriteTimeUtc(cacheFilePath);

            var reinitCommandResult = new DotnetNewCommand(_log, "--debug:rebuildcache")
               .WithCustomHive(home)
               .Execute();

            reinitCommandResult.Should().ExitWith(0).And.NotHaveStdErr();
            Assert.Equal(commandResult.StdOut, reinitCommandResult.StdOut);
            Assert.True(File.Exists(cacheFilePath));
            Assert.True(lastUpdateDate < File.GetLastWriteTimeUtc(cacheFilePath));
        }
    }
}
