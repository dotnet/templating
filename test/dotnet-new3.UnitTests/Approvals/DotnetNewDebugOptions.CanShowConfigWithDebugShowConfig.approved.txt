Current configuration:
 
Mount Point Factories                 Type                                                                        Assembly                                                                                            
------------------------------------  --------------------------------------------------------------------------  ----------------------------------------------------------------------------------------------------
94e92610-cf4c-4f6d-aeb6-9e42dde1899d  Microsoft.TemplateEngine.Edge.Mount.Archive.ZipFileMountPointFactory        Microsoft.TemplateEngine.Edge, Version=<version>, Culture=neutral, PublicKeyToken=<token>
8c19221b-dea3-4250-86fe-2d4e189a11d2  Microsoft.TemplateEngine.Edge.Mount.FileSystem.FileSystemMountPointFactory  Microsoft.TemplateEngine.Edge, Version=<version>, Culture=neutral, PublicKeyToken=<token>


Generators                            Type                                                                             Assembly                                                                                                                     
------------------------------------  -------------------------------------------------------------------------------  -----------------------------------------------------------------------------------------------------------------------------
0c434df7-e2cb-4dee-b216-d7c58c8eb4b3  Microsoft.TemplateEngine.Orchestrator.RunnableProjects.RunnableProjectGenerator  Microsoft.TemplateEngine.Orchestrator.RunnableProjects, Version=<version>, Culture=neutral, PublicKeyToken=<token>


The 'dotnet new3' command creates a .NET project based on a template.

Common templates are:
Template Name  Short Name  Language    Tags          
-------------  ----------  ----------  --------------
Class Library  classlib    [C#],F#,VB  Common/Library
Console App    console     [C#],F#,VB  Common/Console

An example would be:
   dotnet new3 console

Display template options with:
   dotnet new3 console -h
Display all installed templates with:
   dotnet new3 --list
Display templates available on NuGet.org with:
   dotnet new3 web --search